### PR TITLE
Added connection in GCSBatchSink and GCSMultiFileBatchSink

### PR DESCRIPTION
**Pull request by itsankit-google**
[PLUGIN-968](https://cdap.atlassian.net/browse/PLUGIN-968)

### CODE REVIEW

**Review by yaojiefeng — COMMENTED** [review]
These changes LGTM, we will also need to add these sinks to the connector's spec method to let UI know which plugins the spec is corresponding to, please see https://github.com/data-integrations/google-cloud/blob/develop/src/main/java/io/cdap/plugin/gcp/gcs/connector/GCSConnector.java#L158

Since the UI need some changes on getting back to studio, you can test it directly using REST endpoint:
namespaces/system/apps/pipeline/services/studio/methods/v1/contexts/default/connections/<connecton-name>/specification
with request body like 
```
{
  "path": "<path-to-file>",
  "properties": {}
}
```
to ensure these sinks are included in the response

**Comment by yaojiefeng** [comment]
Please also make sure the changes work for preview, and actual pipeline run

**Comment by itsankit-google** [comment]
> These changes LGTM, we will also need to add these sinks to the connector's spec method to let UI know which plugins the spec is corresponding to, please see https://github.com/data-integrations/google-cloud/blob/develop/src/main/java/io/cdap/plugin/gcp/gcs/connector/GCSConnector.java#L158
> 
> Since the UI need some changes on getting back to studio, you can test it directly using REST endpoint: namespaces/system/apps/pipeline/services/studio/methods/v1/contexts/default/connections//specification with request body like
> 
> ```
> {
>   "path": "<path-to-file>",
>   "properties": {}
> }
> ```
> 
> to ensure these sinks are included in the response

Done.

**Review by yaojiefeng — APPROVED** [review]

File: docs/GCS-batchsink.md [hunk 1]
```diff
@@ -26,6 +26,12 @@ Properties
 ----------
 **Reference Name:** Name used to uniquely identify this sink for lineage, annotating metadata, etc.
 
+**Use Connection** Whether to use a connection, if a connection is used,
```
Comment (rrielley76) [inline docs/GCS-batchsink.md:29]: @itsankit-google Can we rewrite as "Whether to use a connection. If a connection is used,"

File: docs/GCS-batchsink.md [hunk 2]
```diff
@@ -26,6 +26,12 @@ Properties
...
+the credentials does not need to be provided.
```
Comment (rrielley76) [inline docs/GCS-batchsink.md:30]: @itsankit-google Can we rewrite as "you do not need to provide the credentials."

File: docs/GCS-batchsink.md [hunk 3]
```diff
@@ -26,6 +26,12 @@ Properties
...
+
+**Connection** Name of the connection to use, should use the macro function ${conn:(connection-name)} to provide.
```
Comment (rrielley76) [inline docs/GCS-batchsink.md:32]: @itsankit-google You no longer write a macro for the name. You Browse connection and select one. The connection name displays in the Connection field. The field isn't editable.  Maybe just say "Name of the connection to use. Project and service account information will be provided by the connection." (The last sentence is in the GCS source docs). For the part about using the macro, how's "You also can use the macro function ${conn:(connection-name)}."

**Pull request merged**